### PR TITLE
citra_qt/multiplayer/lobby: Add "Hide Password Protected Rooms" Filter.

### PR DESCRIPTION
**Pull request by adityaruplaha**
Pretty self explanatory.
Also, I removed an extraneous spacer.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3710)

### CODE REVIEW

File: src/citra_qt/multiplayer/lobby.cpp [hunk 1]
```diff
@@ -221,6 +223,14 @@ bool LobbyFilterProxyModel::filterAcceptsRow(int sourceRow, const QModelIndex& s
         return true;
     }
 
+    // filter by password protected rooms
+    if (filter_password_protected) {
+        QModelIndex password_index =
+            sourceModel()->index(sourceRow, Column::ROOM_NAME, sourceParent);
+        if (sourceModel()->data(password_index, LobbyItemName::PasswordRole).toBool())
+            return false;
```
Comment (zhaowenlan1779) [inline src/citra_qt/multiplayer/lobby.cpp]: (perhaps)
```cpp
if (sourceModel()->data(password_index, LobbyItemName::PasswordRole).toBool()) {
    return false;
}
```

File: src/citra_qt/multiplayer/lobby.h [hunk 2]
```diff
@@ -104,12 +104,14 @@ class LobbyFilterProxyModel : public QSortFilterProxyModel {
 
 public slots:
     void SetFilterOwned(bool);
+    void SetFilterPasswordProtected(bool);
```
Comment (zhaowenlan1779) [inline src/citra_qt/multiplayer/lobby.h]: This should be below ```SetFilterFull```

File: src/citra_qt/multiplayer/lobby.h [hunk 3]
```diff
@@ -104,12 +104,14 @@ class LobbyFilterProxyModel : public QSortFilterProxyModel {
...
     void SetFilterFull(bool);
     void SetFilterSearch(const QString&);
 
 private:
     QStandardItemModel* game_list;
     bool filter_owned = false;
+    bool filter_password_protected = false;
```
Comment (zhaowenlan1779) [inline src/citra_qt/multiplayer/lobby.h]: This should be below ```filter_full```

File: src/citra_qt/multiplayer/lobby.cpp (hunk 1 above)
Comment (adityaruplaha) [reply src/citra_qt/multiplayer/lobby.cpp]: I don't think we use braces for one-liner early returns.

File: src/citra_qt/multiplayer/lobby.h (hunk 2 above)
Comment (adityaruplaha) [reply src/citra_qt/multiplayer/lobby.h]: Why exactly?

File: src/citra_qt/multiplayer/lobby.h (hunk 3 above)
Comment (adityaruplaha) [reply src/citra_qt/multiplayer/lobby.h]: Ditto.

**Comment by adityaruplaha** [comment]
>UI changes require a screenshot, so I captured one for you.
Thanks.

>You see, the window does not look very nice.
But I use a different font style, where it looks quite nice?

**Comment by CaptV0rt3x** [comment]
> You see, the window does not look very nice.

regarding this, what @zhaowenlan1779 probably meant was that the text "Hide Password Protected Rooms" is too long and makes the window look not so good. Shortening it or rewording it would make it better in terms of UI.

**Comment by zhaowenlan1779** [comment]
@CaptV0rt3x @adityaruplaha Yes, that's exactly what I meant.

**Comment by baka0815** [comment]
> Perhaps you should just say 'Hide Password Protected'
Then we should also reword ``Hide Full Rooms`` to ``Hide Full`` to keep it consistent.

**Comment by adityaruplaha** [comment]
Okay, as soon as I get some time, I'll try to update this.

**Pull request closed without merging**